pkg/vm/vm.go

```diff
@@ -136,9 +133,7 @@ func (vm *VirtualMachine) RunStep(hintRunner HintRunner) error {
 	}
 
 	// store the trace before state change
-	if vm.config.ProofMode {
-		vm.Trace = append(vm.Trace, vm.Context)
-	}
+	vm.Trace = append(vm.Trace, vm.Context)
```

Comment:
I wouldn't add the trace every time. Collecting the trace vs not is a performance hit. I would suggest to have a flag in the VM.config (perhaps `get_trace`, `collect_trace` or any name you think is best) and keep the if